services/management/dbaas/components_service.go

```diff
@@ -392,3 +402,61 @@ func setComponent(kc *models.Component, rc *dbaasv1beta1.ChangeComponent) (*mode
 	kc.DisabledVersions = stringset.ToSlice(disabledVersions)
 	return kc, nil
 }
+
+func (c componentsService) InstallOperator(ctx context.Context, req *dbaasv1beta1.InstallOperatorRequest) (*dbaasv1beta1.InstallOperatorResponse, error) {
+	kubernetesCluster, err := models.FindKubernetesClusterByName(c.db.Querier, req.KubernetesClusterName)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	var component *models.Component
+	var installFunc func() error
+	switch req.OperatorType {
+	case pxcOperator:
+		installFunc = func() error {
+			_, err := c.dbaasClient.InstallXtraDBOperator(ctx, &controllerv1beta1.InstallXtraDBOperatorRequest{
+				KubeAuth: &controllerv1beta1.KubeAuth{
+					Kubeconfig: kubernetesCluster.KubeConfig,
+				},
+				Version: req.Version,
+			})
+			return err
+		}
+		component = kubernetesCluster.PXC
+	case psmdbOperator:
+		installFunc = func() error {
+			_, err := c.dbaasClient.InstallPSMDBOperator(ctx, &controllerv1beta1.InstallPSMDBOperatorRequest{
+				KubeAuth: &controllerv1beta1.KubeAuth{
+					Kubeconfig: kubernetesCluster.KubeConfig,
+				},
+				Version: req.Version,
+			})
+			return err
+		}
+		component = kubernetesCluster.Mongod
+	default:
+		return &dbaasv1beta1.InstallOperatorResponse{Status: dbaasv1beta1.OperatorsStatus_OPERATORS_STATUS_NOT_INSTALLED},
+			errors.Errorf("%q is not supported operator", req.OperatorType)
+	}
+
+	if component != nil {
+		// Default version of database could be unsupported be a new operator version.
+		supported, err := c.versionServiceClient.IsDatabaseVersionSupportedByOperator(ctx, req.OperatorType, req.Version, component.DefaultVersion)
+		if err != nil {
+			return &dbaasv1beta1.InstallOperatorResponse{Status: dbaasv1beta1.OperatorsStatus_OPERATORS_STATUS_NOT_INSTALLED},
+				status.Errorf(codes.Internal, "failed to check if default database version is supported: %v", err)
+		}
+		if !supported {
+			return &dbaasv1beta1.InstallOperatorResponse{Status: dbaasv1beta1.OperatorsStatus_OPERATORS_STATUS_NOT_INSTALLED},
+				status.Errorf(codes.Internal, "default database version %s is unsupported by the operator version %s, please change default version.", component.DefaultVersion, req.Version)
```

Comment:
so, if there will be no intersects users won't be able to upgrade operator, right?

Reply:
Yes, first I was setting default database to latest version if current version is not supported by operator we want to install. Now, we fail and user must update default database version to supported one. It's what @denisok decided we want to do https://perconacorp.slack.com/archives/C01JQ9F1EAF/p1626449786129300?thread_ts=1626419178.120300&cid=C01JQ9F1EAF